### DISCUSSION TITLE
high false positive rate

### DISCUSSION THREAD

**Discussion by daskandalis** [talmolab/sleap#1596] [Help!] [answered]
I've been getting very poor performance with false positives due to the tracker picking up background noise. Particularly when no animal is in view, the predictor focuses on any noise in the frame, even though the background is dark and doesn't resemble the training points.

![image](https://github.com/talmolab/sleap/assets/1887822/6e1743de-70b5-4736-87d0-b24f238fe5b7)
This image is the entire field of view.

I've also observed that the model is very easily fooled by salt-and-pepper noise, and will focus on on a single saturated pixel against a dark background even when the animal (and the trained feature) is in view.

Is there a way to diagnose this behavior?

**Comment by talmo** [comment, marked as answer]
Hi @daskandalis,

Thanks for reaching out!

This is indeed a tricky one since it might involve a feature described in #640 that we haven't had a chance to implement yet, but your remarks that it happens even in frames which contains the animals is a little more concerning.

Do you mind sharing:

1. A few more screenshots of the salt-and-pepper sensitivity with animals also in the frame
2. The details of your model configuration from the **Run Training** window
3. How many frames you have labeled and the size of your videos

We'll be able to diagnose more once we have those pieces of information, but I also have a faint suspicion that it could have to do with the video not being [reliably seekable](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format), which can result in weird behavior if the model is being trained to find landmarks on frames that are empty because we seek to the wrong one during training.

If you also have it, a screenshot of the visualizations during training would also be helpful for diagnosis!

Cheers,

Talmo

**Comment by daskandalis** [comment]
It appears you are right, this may have resulted from the seeking issue. Converting from a common high-speed filming proprietary format directly to mp4 caused corrupted frames when viewing the video frame-by-frame. I couldn't understand why the corruption depended on going forward or backward to reach the frame, but your link explains it. At present, my issues _appear_ to be solved by first converting to images and then converting the images to a video container. I will close the issue for now, but I agree that training on true negative labels would be a great addition.
